### PR TITLE
K8SPXC-757 fix recovery documentation

### PR DESCRIPTION
**Pull request by hors**
[K8SPXC-757](https://jira.percona.com/browse/K8SPXC-757) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: source/recovery.rst [hunk 1]
```diff
@@ -145,10 +130,7 @@ which is based on three Percona XtraDB Cluster Pods.
 
    .. code-block:: bash
 
-      $ kubectl exec cluster1-pxc-2 -- mysqld --wsrep_recover
-      $ kubectl exec cluster1-pxc-2 -- sed -i 's/safe_to_bootstrap: 0/safe_to_bootstrap: 1/g' /var/lib/mysql/grastate.dat
-      $ kubectl exec cluster1-pxc-2 -- sed -i 's/wsrep_cluster_address=.*/wsrep_cluster_address=gcomm:\/\//g' /etc/mysql/node.cnf
-      $ kubectl exec cluster1-pxc-2 -- mysqld
+      $ kubectl -n default exec cluster1-pxc-2 -c pxc -- sh -c 'kill -s USR1 1'
```
Comment (delgod) [inline source/recovery.rst:133]: namespace should be specified in all commands or in none
```suggestion
      $ kubectl exec cluster1-pxc-2 -c pxc -- sh -c 'kill -s USR1 1'
```

File: source/recovery.rst [hunk 2]
```diff
@@ -99,43 +99,28 @@ which is based on three Percona XtraDB Cluster Pods.
 
    .. code-block:: bash
 
-      $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do until [[ $(kubectl get pod cluster1-pxc-$i -o jsonpath='{.status.phase}') == 'Running' ]]; do sleep 10; done; kubectl exec cluster1-pxc-$i -- touch /var/lib/mysql/sst_in_progress; done
+      $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do until [[ $(kubectl get pod cluster1-pxc-$i -o jsonpath='{.status.phase}') == 'Running' ]]; do sleep 10; done; kubectl exec cluster1-pxc-$i -c pxc -- touch /tmp/recovery-case; done
```
Comment (delgod) [inline source/recovery.rst:102]: do we still need debug image here? lines 50-90

**Review by delgod — CHANGES_REQUESTED** [review]

File: source/recovery.rst (hunk 2 above)
Comment (hors) [reply source/recovery.rst:102]: support wonts to have it, from our end we can avoid this doc at all.

**Review by delgod — DISMISSED** [review]

File: source/recovery.rst [hunk 3]
```diff
@@ -156,6 +189,8 @@ which is based on three Percona XtraDB Cluster Pods.
 #. Go back *to the previous shell* and return the original Percona XtraDB
    Cluster image because the debug image is no longer needed:
 
+   .. note:: Please make sure that the version of PXC image is the same that it was before bebug image.
```
Comment (delgod) [inline source/recovery.rst:192]:
```suggestion
   .. note:: Please make sure that the version of PXC image is the same that it was before setting the debug image.
```

File: source/recovery.rst [hunk 4]
```diff
@@ -19,8 +19,9 @@ continuously restarting, and generates the following errors in the log::
 The Percona Operator for Percona XtraDB Cluster provides two ways of recovery
 after a full cluster crash.
 
-The Operator is able to provide automatic crash recovery starting from version
-``1.7``. For previous Operator versions, crash recovery can be done
+The Operator is able to provide automatic crash recovery and semi automatic
```
Comment (delgod) [inline source/recovery.rst:22]:
```suggestion
The Operator is providing automatic crash recovery (by default) and semi-automatic
```

File: source/recovery.rst [hunk 5]
```diff
@@ -35,7 +36,32 @@ The default value for this option is ``true``, which means that automatic
 recovery is turned on.
 
 If this option is set to ``false``, automatic crash recovery is not done,
-but manual crash recovery is still possible.
+but semi automatic recovery is still possible.
```
Comment (delgod) [inline source/recovery.rst:39]: should we disable spec.pxc.autoRecovery option?

**Review by delgod — CHANGES_REQUESTED** [review]

File: source/recovery.rst (hunk 5 above)
Comment (hors) [reply source/recovery.rst:39]: please see lines 32-39, it this section I am talking about  this particular case

Comment (fiowro) [inline source/recovery.rst:39]:
```suggestion
but semi-automatic recovery is still possible.
```

File: source/recovery.rst [hunk 6]
```diff
@@ -78,6 +104,13 @@ which is based on three Percona XtraDB Cluster Pods.
 
 #. Change the normal PXC image inside the cluster object to the debug image:
 
+   .. note:: Please make sure that the version of PXC for debug image is the same that is used for cluster now.
```
Comment (fiowro) [inline source/recovery.rst:107]:
```suggestion
   .. note:: Please make sure the Percona XtraDB Cluster version for the debug image matches the version currently in use in the cluster. 
```

File: source/recovery.rst [hunk 7]
```diff
@@ -156,6 +189,8 @@ which is based on three Percona XtraDB Cluster Pods.
 #. Go back *to the previous shell* and return the original Percona XtraDB
    Cluster image because the debug image is no longer needed:
 
+   .. note:: Please make sure that the version of PXC image is the same that it was before setting the debug image.
```
Comment (fiowro) [inline source/recovery.rst:192]:
```suggestion
   .. note:: Please make sure the Percona XtraDB Cluster version for the debug image matches the version currently in use in the cluster.
```

File: source/recovery.rst [hunk 8]
```diff
@@ -35,7 +36,32 @@ The default value for this option is ``true``, which means that automatic
 recovery is turned on.
 
 If this option is set to ``false``, automatic crash recovery is not done,
-but manual crash recovery is still possible.
+but semi automatic recovery is still possible.
+
+In this case you need to get the log from pxc container from all Pods
+using the following command:
+
+   .. code-block:: bash
```
Comment (fiowro) [inline source/recovery.rst:44]:
```suggestion
.. code-block:: bash
```

File: source/recovery.rst [hunk 9]
```diff
@@ -35,7 +36,32 @@ The default value for this option is ``true``, which means that automatic
...
+
+      $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do echo "###############cluster1-pxc-$i##############"; kubectl logs cluster1-pxc-$i -c pxc | grep '(seqno):' ; done
```
Comment (fiowro) [inline source/recovery.rst:46]:
```suggestion
   $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do echo "###############cluster1-pxc-$i##############"; kubectl logs cluster1-pxc-$i -c pxc | grep '(seqno):' ; done
```

File: source/recovery.rst [hunk 10]
```diff
@@ -35,7 +36,32 @@ The default value for this option is ``true``, which means that automatic
...
+   It is cluster1-pxc-1.cluster1-pxc.default.svc.cluster.local node with sequence number (seqno): 18
+   ###############cluster1-pxc-2##############
+   It is cluster1-pxc-2.cluster1-pxc.default.svc.cluster.local node with sequence number (seqno): 19
+
+Now find the Pod with the largest ``seqno`` (it is ``cluster1-pxc-2`` in the
+above example).
+
+Now execute the following commands to start this instance:
+
+   .. code-block:: bash
```
Comment (fiowro) [inline source/recovery.rst:62]:
```suggestion
.. code-block:: bash
```

File: source/recovery.rst [hunk 11]
```diff
@@ -35,7 +36,32 @@ The default value for this option is ``true``, which means that automatic
...
+
+      $ kubectl exec cluster1-pxc-2 -c pxc -- sh -c 'kill -s USR1 1'
```
Comment (fiowro) [inline source/recovery.rst:64]:
```suggestion
   $ kubectl exec cluster1-pxc-2 -c pxc -- sh -c 'kill -s USR1 1'
```

File: source/recovery.rst [hunk 12]
```diff
@@ -78,6 +104,13 @@ which is based on three Percona XtraDB Cluster Pods.
 
 #. Change the normal PXC image inside the cluster object to the debug image:
 
+   .. note:: Please make sure that the version of PXC for debug image is the same that is used for cluster now.
+      How to get PXC image which is used for cluster:
```
Comment (fiowro) [inline source/recovery.rst:108]:
```suggestion
      You can run the following command to find out which Percona XtraDB Cluster image is in use:
```

**Review by fiowro — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Review by fiowro — APPROVED** [review]

**Pull request merged**